### PR TITLE
Post comments to correct repository when on a fork.

### LINKED ISSUE #1195: Can't comment on PR from parent repo from fork
- GitHub Extension for Visual Studio version: 2.3.0.24 (published)
- Visual Studio version: 2015/2017

1. Opened a solution in a forked repo `jcansdale/PullRequestSandbox`.
2. Selected a PR in list of PRs from parent repo
3. Opened a PR file (diff view appeared)
4. Tried to add inline comment to file
5. When I clicked `Comment` button `Not Found` appeared

image

It looks like it's trying to publish the comment to a PR on the forked repo.

image

### PR DESCRIPTION
**Pull request by grokys**
If the user had a forked repo checked out and was viewing the parent repo's PRs, posting a comment was erroneously trying to post the comment to the _fork_ repo, resulting in a "Not Found" exception.

Fixes #1195.

### CODE REVIEW

**Comment by jcansdale** [comment]
Could we log the exception as well as writing `Not Found` to the GUI?

image

**Comment by grokys** [comment]
Good point! Will do.

File: src/GitHub.InlineReviews/ViewModels/CommentViewModel.cs [hunk 1]
```diff
@@ -167,6 +169,7 @@ void DoCancelEdit(object unused)
                 }
 
                 ErrorMessage = message;
+                log.Error("Error posting inline comment.", e);
```
Comment (jcansdale) [inline src/GitHub.InlineReviews/ViewModels/CommentViewModel.cs]: Just an observation. In most places we don't seem ends the message with a `.`.

It is a bit all over the place though. I've even seen one like:
`log.Error("foo {0}. ", ex);`

I guess we should do an audit at some point. 😉

Comment (grokys) [reply src/GitHub.InlineReviews/ViewModels/CommentViewModel.cs]: Actually i started out with no "." but then looked in the log and the messages immediately before it had a closing "." so I added it!

Comment (jcansdale) [reply src/GitHub.InlineReviews/ViewModels/CommentViewModel.cs]: lol. LGTM!

**Comment by meaghanlewis** [comment]
Thanks for the fix @grokys 🎉 and for adding unit tests and logging on this! ✅

**Pull request merged**